man/data.table.Rd

```diff
@@ -152,6 +152,10 @@ data.table(\dots, keep.rownames=FALSE, check.names=FALSE, key=NULL, stringsAsFac
     Inversion (column dropping instead of keeping) can be accomplished be prepending the argument with \code{!} or \code{-} (there's no difference between these), e.g. \code{.SDcols = !c('x', 'y')}.
 
     Finally, you can filter columns to include in \code{.SD} based on their \emph{names} according to regular expressions via \code{.SDcols=patterns(regex1, regex2, ...)}. The included columns will be the \emph{intersection} of the columns identified by each pattern; pattern unions can easily be specified with \code{|} in a regex. You can filter columns on \code{values}  by passing a function, e.g. \code{.SDcols=\link{is.numeric}}. You can also invert a pattern as usual with \code{.SDcols=!patterns(...)} or \code{.SDcols=!is.numeric}.
+
+    Note that if there is a user-defined function named \code{patterns}, it will be used instead of the usual patterns, and it should return a list of column names to intersect, and if that function has an argument named \code{cols}, then that argument will be set to \code{names(x)} by non-standard evaluation.
+
+    In addition, \code{.SDcols=fun_not_named_patterns()} is possible, and that function should return a character vector of column names, and non-standard evaluation is not used, even if there is an argument named \code{cols}.
```

Comment:
This is documentation of the current behavior of .SDcols and patterns, which I think is pretty confusing.
Currently, the function name patterns is a special case, for which we use NSE to fill in the cols argument.
But if the user does .SDcols=any_name_other_than_patterns(), then NSE is not used to fill in the cols argument of that function, which is inconsistent and potentially confusing. 
For consistency, simplicity, and to make the documentation easier to understand, I would suggest modifying the code so that patterns() is no longer a special case, and NSE is always used (with any function name).
I'm not sure that would be a "breaking change" since this is currently un-documented and un-tested behavior.
Any comments/opinions/feedback?